src/components/StatusUpdateForm.js

```diff
@@ -29,8 +29,7 @@ const ImageState = Object.freeze({
   UPLOAD_FAILED: Symbol('upload_failed'),
 });
 
-const urlRegex = /(https?:\/\/[^\s]+)/gi;
-
+const urlRegex = /(?:\s|^)((?:https?:\/\/)?(?:[a-z0-9-]+(?:\.[a-z0-9-]+)+)(?::[0-9]+)?(?:\/(?:[^\s]+)?)?)/g;
```

Comment:
Thanks to @AnatolyRugalev !! Works pretty well - https://regex101.com/r/3ms9WF/1/